node/src/chain_spec.rs

```diff
@@ -617,6 +625,12 @@ fn amplitude_genesis(
 			batching_api: b"https://dia-00.pendulumchain.tech/currencies".to_vec(),
 			coin_infos_map: vec![],
 		},
+		token_allowance: Default::default(),
+		transaction_payment:  Default::default(),
+		treasury:  Default::default(),
+		treasury_buyout_extension: Default::default(),
+		vesting: Default::default(),
+		zenlink_protocol: Default::default(),
```

Comment:
I found the [implementation of the construct_runtime ](https://paritytech.github.io/polkadot-sdk/master/src/frame_support_procedural/construct_runtime/mod.rs.html) because I wanted to see if the implicit declaration exposes everything by default. If I understand correctly, this is defined by the `tt_default_parts()` function that is automatically implemented on each pallet through the `#[pallet:pallet]` macro, see [here](https://paritytech.github.io/polkadot-sdk/master/src/frame_support_procedural/construct_runtime/mod.rs.html#35-36). I don't want to go further down the rabbit hole but this indicates that the implicit declaration might introduce unwanted side effects that are hard to debug (?). After all, the default genesis is [still available](https://github.com/pendulum-chain/pendulum/blob/7f9d96d1e0e7a7eebc46e034287bb24e733d93f5/pallets/orml-currencies-allowance-extension/src/lib.rs#L105-L123). @gianfra-t are you sure that the need for all these `Default::default()` statements arose after changing the definitions in the `construct_runtime` macro from explicit to implicit? Or could it be related to the version update?

If this really is required because we changed to the implicit definitions, then I wonder what else this entails that is not obvious and I'm a little scared that this gives us a headache later.

Reply:
So I actually tried this. I rollback on one of the definitions, in this case `security` like so: 
```
Security: security::{Pallet, Call, Config<T>, Storage, Event<T>} = 67,
```
and I still get 

```
Security: security::{Pallet, Call, Config<T>, Storage, Event<T>} = 67,
```

This means that the explicit/implicit declaration does not really affect this new requirements for genesis. 
But, for the pallets that now need to add `Default()`, then declaring it explicitly without `Config` will indeed remove this requirement.

So, re-adding 
```
TokenAllowance: orml_currencies_allowance_extension::{Pallet, Storage, Call, Event<T>} = 80,
```
even forbids the use of declaring `token_allowance: Default::default(),` in genesis.

I don't know if this adds much information, at the end of the day, we still don't know exactly extra consequences could have leaving them implicit. I will try to find more info outside of reading the macro declaration.

Reply:
@TorstenStueber do you have any thoughts about changing from the explicit declaration to the implicit one?

Reply:
I used `cargo expand` to expand `construct_runtime` using the explicit (with `::{Pallet, Call, Storage, Event<T>}`) and the implicit definition for `pallet_vesting`. Then I diffed the result. The main difference (apart from a long list of definitions from serde) is that with the implicit definition the following type is defined:

```
pub type VestingConfig = pallet_vesting::GenesisConfig<Runtime>;
```

and the `pub struct RuntimeGenesisConfig` has an additional entry `pub vesting: VestingConfig`.

When using the explicit definition with `::{Pallet, Call, Config<T>, Storage, Event<T>}` (i.e. I added `Config<T>`) instead, then the result is equal to the implicit definition.

So I think the only difference in behavior between explicit and implicit definition affects all pallets where we did not explicitly state `Config<T>` before. I assume that this is the only side effect.

I think it is nicer to have a nice `construct_runtime` with implicit definitions and then live with stating all the `Default::default()` in the chain spec.

Reply:
@ebma I used `cargo expand` to expand `construct_runtime` using the explicit (with `::{Pallet, Call, Storage, Event<T>}`) and the implicit definition for `pallet_vesting`. Then I diffed the result. The main difference (apart from a long list of definitions from serde) is that with the implicit definition the following type is defined:

```
pub type VestingConfig = pallet_vesting::GenesisConfig<Runtime>;
```

and the `pub struct RuntimeGenesisConfig` has an additional entry `pub vesting: VestingConfig`.

When using the explicit definition with `::{Pallet, Call, Config<T>, Storage, Event<T>}` (i.e. I added `Config<T>`) instead, then the result is equal to the implicit definition.

So I think the only difference in behavior between explicit and implicit definition affects all pallets where we did not explicitly state `Config<T>` before. I assume that this is the only side effect.

I think it is nicer to have a nice `construct_runtime` with implicit definitions and then live with stating all the `Default::default()` in the chain spec.

Reply:
Cool, thanks for confirming @TorstenStueber 👍 I agree that the implicit version is more convenient and I'm also happy to switch to that. Let's keep it the way you changed it then @gianfra-t.

Reply:
Good idea to test the expansion @TorstenStueber, didn't think of that. Let's keep the implicit.